### PR TITLE
Media support and lots of other things

### LINKED ISSUE #97: User profile view
Create first-pass user profile view with:

### Requirements
- [x] Store remote user data in _local_ and _remote_
- [ ] Inputs
- [ ] Tabs
- [x] Handle redirect to section
- [ ] Password reset
- [ ] Update local on input change

### Fields/tabs

**Account tab**
- [ ] Full name
- [ ] Email
- [ ] A bio
- [ ] Profile image

**Settings**
- [ ] Language
- [ ] Timezone
- [ ] Date/time format

**Password reset**
- [ ] Old password
- [ ] New password
- [ ] Repeat new password

### LINKED ISSUE #170: Move network calls to actions
As per discussion, this is to be implemented: https://github.com/dadi/publish/blob/master/docs/stores.md

### LINKED ISSUE #175: Update API status to idle only when collections load

### PR DESCRIPTION
**Pull request by eduardoboucas**
This PR:

- Ensures API status is only set to idle once all collections have been loaded (closes #175)
- Moves remaining network calls to actions (closes #170)
- Loads field components dynamically and introduces the concept of [contexts](https://github.com/dadi/publish/blob/feat/media/docs/field-components.md)
- Hides auth and media collections from collection nav
- Decouples toolbar components from `DocumentEdit` container. Actions moved to the toolbar, leaving the container with the sole purpose of displaying fields and register changes, which allows it to be reused in multiple contexts (e.g. profile edit view)
- Implements profile edit view (closes #97)
- Adds media support
- A lot of small things (see change log 🎉 🍾 😅)

### CODE REVIEW

File: frontend/components/ButtonWithOptions/ButtonWithOptions.jsx [hunk 1]
```diff
@@ -112,13 +118,13 @@ export default class ButtonWithOptions extends Component {
           onClick={this.toggleOptions.bind(this)}
         >
           <IconArrow
-            direction={this.state.optionsExpanded ? 'down' : 'up'}
+            direction={optionsExpanded ? 'down' : 'up'}
```
Comment (mingard) [inline frontend/components/ButtonWithOptions/ButtonWithOptions.jsx]: Why not just `open`? 
`direction={open ? 'down' : 'up'}`

Comment (eduardoboucas) [inline frontend/components/ButtonWithOptions/ButtonWithOptions.jsx]: You mean changing the state variable name from `optionsExpanded` to `open`? Just making sure, as I haven't touched that.

Comment (mingard) [inline frontend/components/ButtonWithOptions/ButtonWithOptions.jsx]: Yes. Do you agree?

Comment (eduardoboucas) [inline frontend/components/ButtonWithOptions/ButtonWithOptions.jsx]: I don't mind either way. I'll change shortly.

Comment (mingard) [inline frontend/components/ButtonWithOptions/ButtonWithOptions.jsx]: Sweet

**Pull request merged**